meta-balena-common/recipes-connectivity/networkmanager/resin-files/98dhcp_ntp

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+
+#
+# Update NTP sources that are passed via DHCP.
+#
+
+INTERFACE=$1
+ACTION=$2
+
+SERVER_DIR=/run/chrony/
+SERVER_FILE=${SERVER_DIR}/added_dhcp_${INTERFACE}.sources
+
+if [ ! -d "$SERVER_DIR" ]; then
+	mkdir -p $SERVER_DIR
+	chmod 750 $SERVER_DIR
+fi
+
+case "$ACTION" in
+	up|dhcp4-change)
```

Comment:
Why did you ditch the connectivity-change event? If we only use the on/off events we only react to interface on/off events, but loss of connectivity of the type the network connectivity check does will not trigger NTP/DHCP updates. In the time an upstream router disconnects us from the network lease may have expired and NTP servers may have failed.

Reply:
If you look at the NM documentation [here](https://developer.gnome.org/NetworkManager/unstable/NetworkManager.html) you will see that 'connectivity-change' is generic, i.e. the interface name passed to the dispatcher script is empty. I can't find any evidence to suggest that DHCP NTP sources are passed on this event. I've also tested these changes with a local DHCP server and reached the same conclusion. The example 'chrony.nm-dispatcher.dhcp' script in the chrony repository also only uses 'up', 'dhcp4-change' and 'down'. Note that 'dhcp4-change' covers lease renew/bound events.

Looking at the old script if there was a case where DHCP NTP servers were set on a 'connectivity-change' then a blank interface would cause an entry similar to the following in the servers file: '=0.ubuntu.pool.ntp.org'. Based on the fact that 'down' events have an associated interface then it would try to remove 'eth0=0.ubuntu.pool.ntp.org' which would fail.

Reply:
Could you please test the case where internet is lost upstream and the lease expires?